lib/utils/branches.js

```diff
@@ -49,6 +50,31 @@ function create(repoState, driver, name, opts) {
 }
 
 /**
+ * Fetch the list of branches, and update the given branch only. Will update
+ * the WorkingState of the branch (and discard previous
+ * @param {RepositoryState} repoState
+ * @param {Driver} driver
+ * @param {Branch | String} branchName The branch to update
+ * @return {Promise<RepositoryState>} with the branch updated
+ */
+function update(repoState, driver, branchName) {
+    branchName = Normalize.branchName(branchName || repoState.getCurrentBranch());
+
+    return driver.fetchBranches()
+    .then(function (branches) {
+        var newBranch = branches.find(function (b) {
+            return b.getFullName() === branchName;
+        });
+
+        if (!newBranch) {
+            return repoState;
+        } else {
```

Comment:
The `else` statement is not necessary if you early return the `repoState` in the above condition. It is up to you but you can gain one level of indentation.